app/models/document.rb

```diff
@@ -126,6 +130,11 @@ def humanized_document_type
     document_type.underscore.tr("_", " ")
   end
 
+  def edition_versions
+    Version.where(item_type: "Edition", item_id: editions.select(:id))
+           .order(created_at: :asc, id: :asc)
```

Comment:
not a blocker and i'm guessing the reason, but why do we need to order by both of these? Would we ever get a record where the created at and id are out of sync?

Reply:
This replicates the same sorting as is already defined on `Edition#versions` here:

https://github.com/alphagov/whitehall/blob/d4f946000a26adbf375e95e6e2bc771cdcf5622d/app/models/edition/audit_trail.rb#L17

Versions will primarily be sorted by their `created_at` timestamp. However the `id` is used as a tie-breaker, which is possible because it's an auto-incrementing number so newer ones will always be higher. So the secondary sort on `:id` just protects against that edge case. Without it, we could end up with an indeterminate sort order which could cause history to sort unpredictably in those cases.

I believe the resolution of the `created_at` timestamp field is one second, which is the [default for MySQL `DATETIME` fields](https://dev.mysql.com/doc/refman/8.0/en/fractional-seconds.html). It's not likely that the edition will be updated multiple times within the same second, but it's not impossible. There could be a race condition between users hitting 'save', or even a script which loops and updates things more than once, etc.

In practice, I don't think this is likely to happen. But it's actually pretty common in our test suite. We regularly call `Timecop.freeze` to stop time, and then sequentially create new versions of an edition to manipulate it into the desired state.

For example, here's what `publication.versions` looks like [in this test](https://github.com/alphagov/whitehall/blob/d4f946000a26adbf375e95e6e2bc771cdcf5622d/test/unit/edition_test.rb#L291-L299):

<img width="1595" alt="Screenshot 2022-06-13 at 17 15 36" src="https://user-images.githubusercontent.com/7735945/173398531-712fc352-dad1-4e53-9c7a-7e22dfd9133e.png">

Note that the `created_at` timestamps are all identical. So we've implicitly come to depend upon on that secondary sort acting as a tie-breaker.